### PR TITLE
NN-3589 fix connection proxy issues

### PR DESCRIPTION
**Pull request by JamieDraperUK**
There seems to be a problem attaining an Oracle proxy connection. The username doesn't seem to exists and we think this is because the verify access checks are being triggered on the read only replication database where this user doesn't exists. Moving these checks to the service should ensure that the correct db is checked as the whole service is within a write transaction.

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 1]
```diff
@@ -123,7 +122,7 @@ public InmateDetail getOffender(@Pattern(regexp = "^[A-Z]\\d{4}[A-Z]{2}$", messa
         @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse.class)})
     @ApiOperation("Full details about the current state of an offender")
     @GetMapping("/{offenderNo}/prison-timeline")
-    @VerifyOffenderAccess(overrideRoles = {"SYSTEM_USER", "VIEW_PRISONER_DATA"})
+    @PreAuthorize("hasAnyRole('SYSTEM_USER', 'VIEW_PRISONER_DATA')")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:125]: will need to ponder this - these annotations are doing completely different things:
`VerifyOffenderAccess` allows all users to call the endpoint with overrides for those two roles if the user doesn't have privileges
`PreAuthorize` requires that the users must have those two roles in order to access the endpoint.

Need to decide on an endpoint by endpoint basis which is correct I fear

Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:125]: I've just moved to @VerifyOffenderAccess to the service layer and pulled @PreAuthorize up to the controller level for consistency

File: src/main/java/uk/gov/justice/hmpps/prison/service/MovementsService.java [hunk 2]
```diff
@@ -115,7 +116,7 @@ public MovementsService(final MovementsRepository movementsRepository,
                 .build());
     }
 
-    @PreAuthorize("hasAnyRole('SYSTEM_USER', 'VIEW_PRISONER_DATA')")
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/service/MovementsService.java:118]: Moved to the controller

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**